### DISCUSSION TITLE
Mototrbo

### DISCUSSION THREAD

**Discussion by Heijb** [OK-DMR/ok-dmrlib#4] [Q&A]
Good afternoon,
cool piece of software you have made. Unfortunately I can't figure out how to get the software to work on windows. With a MotoTrbo DMR radio. I am especially interested in ARS and LRRP.
Maybe you can help me out.

Would love to hear from you

regards bas

**Comment by smarek** [comment]
Hey @Heijb 
which piece of software you have in mind? If you mean [ok-dmrlib](https://github.com/OK-DMR/ok-dmrlib), it's protocol encoding and decoding library.

Basically you need to find a place, where you are able to obtain DMR bursts, LRRP/ARS/TMP/... payloads/PDUs, and then you can use ok-dmrlib to decode the data. And other way around obviously, if you can put your data into the repeater/radio, you can encode location request into LRRP PDU using ok-dmrlib.

Let me know how you would like to use it and maybe I can figure something out for you

**Reply by Heijb** [reply]
Hi @smarek 

Thanks for your quick reply.
Im a code-noob, just start coding so it is a little lik~e magic you tell me.

My situation at this moment,
I have my radio's setup with the right ID's.
And i have a script to visualize the locations on OpenStreetMap with Leaflet.

But i am a little lost in the code for getting the location from the Radio into my script.

My idea the staps that must happen:
- Receive the ARS message
- Reply on the ARS with a location(LRRP) request with a interval (for example  60 seconds)
- Receive the LRRP message
- Decode the LRRP message
- put it to my leaflet-map      (This part can i build if i get a  Time, Lat, Lon)

My setup is that i have MOTROBO mobile radio connected to USB.
I see that the ARS message is received on my radio (i see the receive LED's flash on startup of my other radio).

Can you help me a stap further how i can code the staps needed to get the LRRP working?
I'm new to coding so no idea how to start.

**Reply by smarek** [reply]
OK, i've never got Mototrbo radio connected over USB serving the TRBO protocols stack, so I don't know how it works.
I will need Mototrbo ADK documents related to this.

Hytera has similar feature (UART/TTL over USB), where first application protocol is HRNP which carries all the other protocols (TMP, LP, RRS, ...). You just need a special usb to "accessory interface" and documentation of the protocols used.

I found this pdf https://www.qsl.net/kb9mwr/projects/dv/mototrbo/MOTOTRBO%20Application%20Development%20Kit%20Overview.pdf
showcasing several ways to connect Mototrbo radio over USB to either "IP Capable" or "non-IP Capable" peripherial

![2023-08-01-195353_1530x742_scrot](https://github.com/OK-DMR/ok-dmrlib/assets/384824/526d7a7a-ea28-4969-8858-04af7aa877bd)
![2023-08-01-195452_1369x564_scrot](https://github.com/OK-DMR/ok-dmrlib/assets/384824/178bef1c-01c6-4d6d-a3a8-add0e2e24367)

XNL and XCMP are protocols I don't have specification for, and I'll need it if I am to proceed

Also there is mention of "LBS Application" in section 2.5 MOTOTRBO Location Data ADK , see
![2023-08-01-195814_995x629_scrot](https://github.com/OK-DMR/ok-dmrlib/assets/384824/1e9a70b3-b72f-4c61-82d3-9da9033c6098)

Which requires far less protocols to implement (LRRP will be served over UDP, not UART/TTY). This seems feasible to implement to me, but let me know, if the described communication schema fits your needs.

I'd implement LBS, if it was only matter of extracting the terminal position, otherwise you'll need more data services and protocols, then I'd choose IP capable peripherial mode, however ADK is not provided for free (i investigated and did not pursue, since I don't have that kind of budget).

Let me know your thoughts on this, i have trbo handhelds and i can work with RNDIS on linux (my main) and windows (your main), so LBS might not be that complicated to get.
